### DISCUSSION TITLE
Is Volar specially for TypeScript?

### DISCUSSION THREAD

**Discussion by ngekoding** [vuejs/language-tools#751] [Q&A] [answered]
Sorry for a basic question.
I just want to make sure is Volar specially for Vue with TypeScript or can be used for Javascript (.js)?

Thanks in advance

**Comment by johnsoncodehk** [comment, marked as answer]
Volar can use with js, but it can only provide limited intellisense in template.

**Reply by ngekoding** [reply]
So it is very recommended to use TS instead?

**Reply by johnsoncodehk** [reply]
The tool is not responsible for recommending whether users use TS, it depends on your expectations of type-check and IDE intellisense.
You only need to choose whether to use JS or TS based on your experience. Switching to vue/volar should not affect your choice.

Just my personal suggestion. Even if you choose to use JS, TS is still a skill you must master.
